scenario/constraints/NativeTokenConstraint.ts

```diff
@@ -0,0 +1,27 @@
+import { Constraint } from '../../plugins/scenario';
+import { CometContext } from '../context/CometContext';
+import { Requirements } from './Requirements';
+
+export class NativeTokenConstraint<T extends CometContext, R extends Requirements> implements Constraint<T, R> {
+  async solve(_requirements: R, _context: T) {
+    return [
+      async function (ctx: T): Promise<T> {
+        for (const symbol in ctx.assets) {
+          const contract = await ctx.world.deploymentManager.contract(symbol);
+          if (contract.deposit) {
+            const fauceteer = await ctx.getFauceteer();
+            const amount = 200_000n * 10n**18n;
```

Comment:
nit: `exp()`?

Reply:
Not sure its worth the import?

Reply:
But yeah, and we should probably call `contract.decimals()` too